change(tinyusb): Added repo templates for issue and pull request

roma-jam opened the pull request:
**Describe the PR**
Added the templates to: 
- Bug Report

and Pull Requests: 

- Update Component

**Additional context**
To simplify the PR workflow and keep all PRs in the same format.

roma-jam committed:
change(tinyusb): Added repo templates for issue and pull request

tore-espressif commented on .github/ISSUE_TEMPLATE/bug-report.yml:
```diff
@@ -0,0 +1,62 @@
+      label: Which component are you using? If you choose Other, provide details in More Information.
+      multiple: false
+      options:
+        - device/esp_tinyusb
+        - host/cdc
+        - host/hid
+        - host/msc
+        - host/uvc
+        - Other
```
Should this be in esp-usb repository?

roma-jam replied:
I removed that part. 
Thanks for checking it.

peter-marcisovsky commented on .github/ISSUE_TEMPLATE/bug-report.yml:
```diff
@@ -0,0 +1,62 @@
+          required: true
+        - label: I am using target and esp-idf version as defined in component's idf_component.yml
+          required: true
+        - label: I have searched the [issue tracker](https://github.com/espressif/esp-usb/issues?q=is%3Aissue) for a similar issue and not found any related issue.
```
This link is for `esp-usb` github issues

roma-jam replied:
espressif/tinyusb repo doesn't have the "Issues" itself. 

I asked to add them and changed the link accordingly. 
Thanks.

peter-marcisovsky commented on .github/ISSUE_TEMPLATE/feature-request.yml:
```diff
@@ -0,0 +1,33 @@
+    attributes:
+      value: |
+        * We welcome any ideas or feature requests! It’s helpful if you can explain exactly why the feature would be useful.
+        * There are usually some outstanding feature requests in the [existing issues list](https://github.com/espressif/esp-usb/labels/Type%3A%20Feature%20Request), feel free to add comments to them.
```
Here as well, link to `esp-usb` github

roma-jam replied:
I removed this template as I don't see it can be helpful.
Just based on the rate how often we get such feature-requests in espressif/tinyusb.

roma-jam commented:
@tore-espressif @peter-marcisovsky 
Thanks for such a quick review, but I basically just copied files from the esp-usb, because I wanted to check it.
Now, while I still have concerns, I will request for the review when it will be ready.

roma-jam commented:
@tore-espressif @peter-marcisovsky 
I have resolved the problem with Pull Request template (nested templates are not working ATM) and removed the issues templates, which I think we do not need. 

PTAL again.

peter-marcisovsky commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -1,86 +1,48 @@
+          required: true
+        - label: I am using target and esp-idf version as defined in component's idf_component.yml
+          required: true
+        - label: I have searched the [issue tracker](https://github.com/espressif/tinyusb/issues?q=is%3Aissue) for a similar issue and not found any related issue.
```
I suppose, the `Issues` tab will be created in this repo after merging this changes, as right now the tab is missing and this link gets you to `Pull requests` tab.

roma-jam replied:
No, Issues tab should be created by maintainers. 

So, the questions is: Do we want split the issues for espressif/tinyusb? 

Basically, right now, there are three options: 
1. Use `esp-usb` issue tracker
2. Use `esp-idf` issue tracker
3. Create and use `espressif/tinyusb` issue tracker

I guess that this questions is being asked all the time, when new commiter comes.
Anyway, @tore-espressif do we have any special requirements why we didn't create the issue tracker for `espressif/tinyusb`?
Also, please, share your opinion about question which tracker to use. Thanks.

peter-marcisovsky replied:
Oh sry, I haven't thought about the issues tracker in the tinyusb upstream. 

I took a look at the other espressif github repos, which are forked, just like espressif/tinyusb, [here](https://github.com/espressif?q=&type=fork&language=&sort=). 

Some espressif forks have the issue trackers, some don't. Sometimes, even the upstream repo does not have the issue tracker, so the espressif fork has to have one. Looks like, it's rather a personal choice to have one.

peter-marcisovsky commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -1,86 +1,48 @@
-        2. Click on '....'
-        3. See error
+      label: Development Kit.
+      description: On which Development Kit does this issue occur on?
```
I think, here should be either 
```suggestion
      description: Which Development Kit does this issue occur on?
```

or

```suggestion
      description: On which Development Kit does this issue occur?
```

peter-marcisovsky commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -1,86 +1,48 @@
-        Note2: To enable logging, add `LOG=2` to to the make command if building with stock examples or set `CFG_TUSB_DEBUG=2` in your tusb_config.h.
-        More information can be found at [example's readme](https://github.com/hathach/tinyusb/blob/master/docs/getting_started.md)
+      label: Used Component version.
+      description: On which Component version does this issue occur on? Check `dependencies.lock` file in your project root to find it.
```
Likewise here.. slight redundancy of 'on'

peter-marcisovsky commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -1,86 +1,48 @@
-        - Add your-own-example based on stock one
-        - Push and post it here.
+      label: ESP-IDF version.
+      description: On which ESP-IDF version does this issue occur on? Run `git describe --tags` in your esp-idf folder to find it.
```
Likewise here.. slight redundancy of 'on'

peter-marcisovsky commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -1,86 +1,48 @@
-      label: Screenshots
-      description: If applicable, add screenshots to help explain your problem.
+      label: More Information.
+      description: Do you have any other information from investigating this?
```
Optionally: we can add a word issue, or a bug.. or something similar. Just to be sure  
```suggestion
      description: Do you have any other information from investigating this issue?
```

peter-marcisovsky commented on .github/PULL_REQUEST_TEMPLATE.md:
Shouldn't the `PULL_REQUEST_TEMPLATE` be a folder name, inside of which an whatever `*.md` file resides? Just looking at the esp-usb repo, where we have it like this.

roma-jam replied:
ATM, the nesting templates for pull requests are not working. 
At least that what I was able to find and after removing the folder everything started to work well. 

Also, there was a `pull_request_template.md` in the `.github` folder, but I decided to keep it with uppercase names, as when there are lots of files it can be easily missed.

peter-marcisovsky replied:
Do you think, it might be the same issue with the magic PR template we are seeing in esp-usb? So we will just have to rename/restructure the `.md` files, for our PR template to appear?

roma-jam replied:
@peter-marcisovsky IDK, I haven't check it in esp-usb, but for other my personal repo it worked.

roma-jam commented:
@tore-espressif @peter-marcisovsky 
Thanks for the reviewing. 
I have updated the links to issues trackers (re-direct the user to upstream issues or to esp-usb issue tracker) and also added the `issues`  optional field to `idf_component.yml`

PTAL.

peter-marcisovsky reviewed (approved):
Thx for the changes. LGTM
